test/league_stats_test.rb

```diff
@@ -0,0 +1,101 @@
+require_relative 'test_helper'
+require './lib/game_statistics'
+require './lib/stat_tracker'
+require './lib/league_stats'
+
+class LeagueStatisticsTest < Minitest::Test
+  def setup
+    game_path = './data/games.csv'
+    team_path = './data/teams.csv'
+    game_teams_path = './data/game_teams.csv'
+
+    locations = {
+      games: game_path,
+      teams: team_path,
+      game_teams: game_teams_path
+    }
+
+    @stat_tracker = StatTracker.from_csv(locations)
+    @raw_teams_stats = @stat_tracker.teams_stats
+    @raw_game_stats = @stat_tracker.game_stats
+    @raw_game_teams_stats = @stat_tracker.game_teams_stats
+    @game_statistics = GameStatistics.new(@raw_game_stats, @raw_game_teams_stats)
+    @league_stats = LeagueStatistics.new(@raw_teams_stats, @game_statistics)
+  end
+
+  def test_it_exits
+    assert_instance_of LeagueStatistics, @league_stats
+  end
+
+  def test_attributes
+    assert_equal 32, @league_stats.teams_data.length
+  end
+
+  def test_count_of_teams
+    assert_equal 32, @league_stats.count_of_teams
+  end
+
+  def test_group_by_team_id
+    assert_equal 32, @league_stats.group_by_team_id.keys.count
+  end
+
+  def test_team_id_and_average_goals
+    assert_equal 32, @league_stats.team_id_and_average_goals.count
```

Comment:
These tests look like a great place to use mocks and stubs! Should we look into doing that when refactoring?